sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py

```diff
@@ -235,10 +249,148 @@ def set_routes(
         ]
         self._rest_service.sip_routing.update(body=SipConfiguration(routes=routes_internal), **kwargs)
 
+    @distributed_trace
+    def test_routes_with_number(
```

Comment:
Feedback from review:
Change the function def to:
def get_routes_for_number(self, target_phone_number: str, test_routes: SipTrunkRoute[], **kwargs: Any): # type: (...) -> SipTrunkRoute[]

Reply:
So, internally map the test_routes to sipConfiguration and send it to API. And when receiving the response from API, map the resulting RoutesForNumber to SipTrunkRoute[] and return only that.